### PR TITLE
Add ZipGenerator

### PR DESCRIPTION
**Pull request by MattTaylorDLS**
Zip Generator which combines specific generators into one

### CODE REVIEW

File: scanpointgenerator/generators/zipgenerator.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        self.generators = AGenerators([deserialize_object(g, Generator)
+                                      for g in generators])
+
+        units = []
+        axes = []
+        size = self.generators[0].size
+
+        for generator in self.generators:
+            assert generator.axes not in axes, "You cannot zip generators " \
+                                               "on the same axes"
```
Comment (xen-0) [inline scanpointgenerator/generators/zipgenerator.py]: I think, in general, I have not aligned when splitting over multiple lines in this code, but just used a double-indent.

Comment (MattTaylorDLS) [reply scanpointgenerator/generators/zipgenerator.py]: I've aligned here to stop pycharm complaining about a PEP-8: continuation line ocer-indented for visual indent

**Comment by MattTaylorDLS** [comment]
Not sure about the alternate. Thinking about it, I guess the component generators should be checked, and if they all alternate, the parent zipgenerator should be set to alternate, and if they all don't alternate then the parent shouldn't alternate. If some alternate and some dont, maybe it should throw an assertion error?

**Comment by MattTaylorDLS** [comment]
Tom agreed - now takes into account component alternate values

**Comment by MattTaylorDLS** [comment]
New behaviour changed to have alternate set on init to ZipGenerator only. If alternate is set to True on any of the component generators, it will fail. OK for everyone?

**Pull request merged**